### ISSUE TITLE
Animation flickering/trembling on mobile

### ISSUE THREAD

**Issue by jarekdc** [playcanvas/engine#1267]
I’m having an issue on mobile devices where the animation flickers, when running on a PC the animation is pretty smooth but when testing the same setup on iPad or Samsung S8 there is a noticeable flicker happening.

There’s no code added, only the model and an “idle” clip on it.

I assume the problem lies with the way the model / animation was prepared / exported, the generated model .json file's size is: 32.2 MB.

Here’s the link to the editor:
https://playcanvas.com/editor/scene/625770

And a live build:
https://playcanv.as/p/GCxiY7lI/

WebGL Report:
![image](https://user-images.githubusercontent.com/11174669/43140879-e8cc3baa-8f54-11e8-96d0-ff8126f92e6f.png)

![image](https://user-images.githubusercontent.com/11174669/43140902-f301c3b0-8f54-11e8-8585-e0e2c0161548.png)

1. Open the live build link on mobile device and on a PC.
2. Notice the issue with the animation flickering/trembling on mobile device compared to it being smooth on PC.

**Comment by willeastcott** [comment]
I believe that this is caused by performing skinning in world space. It should probably be updated to be done in the local space of the skinned mesh instance.

**Comment by mvaligursky** [comment]
I did some investigations on this.

In current engine implementation, skinning matrices are supplied in world space. For example if engine with SkinnedMesh has 500k position, all bones matrices will have that offset in them.
I hacked the code to remove this from bone matrixes, effectively doing skinning in local space in the vertex shader, and then adding the offset back in using model matrix. This did not improve the end result, and the character was wobbling the same. We either introduce float precision error when computing bone matrices, or we introduce it in vertex shader when adding model matrix offset to locally skinned position, but in both cases we do something like 500k + 2 (2 being the skinning contribution), which results in precision loss in a value we output from vertex shader.
